src/sources/commondatamodel.jl

```diff
@@ -385,12 +385,16 @@ end
 function _cdmspan(index, order)
     # Handle a length 1 index
     length(index) == 1 && return Regular(zero(eltype(index))), Points()
-    
-    # Calculate step, avoiding as many floating point errors as possible
-    st = Float64(Base.step(Base.range(first(index), last(index); length = length(index))))
-    st_rd = round(st, digits = Base.floor(Int,-log10(eps(eltype(index))))) # round to nearest digit within machine epsilon
-    step = st_rd ≈ st ? st_rd : st # keep the rounded number if it is very close to the original
 
+    step = if eltype(index) <: AbstractFloat
+        # Calculate step, avoiding as many floating point errors as possible
+        st = Float64(Base.step(Base.range(first(index), last(index); length = length(index))))
```

Comment:
```suggestion
        st = Base.step(Base.range(Float64(first(index)), Float64(last(index)); length = length(index)))
```

The `Float64` conversion needs to happen before the division in `range`, not after